Constraint Manager now available for Bounds Control

thalbern opened the pull request:
## Overview
PR integrates Constraint Manager into Bounds Control. 

- removed hardcoded min max scale constraint
- bounds control now supports scale and rotation constraints (translation constraints to be added once translation handles are in)
- adjusted min max scale constraint test and added test to verify rotation constraint
- adjusted constraint manager to check for added constraints during runtime (old design only allowed to have constraints registered that were added before the component that includes the constraint manager)

## Changes
- Fixes: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/8304
part of #5340

thalbern commented:
/azp run

thalbern commented:
/azp run

wiwei commented on Assets/MRTK/SDK/Features/Input/Handlers/Constraints/ConstraintManager.cs:
```diff
@@ -59,5 +65,25 @@ private void ApplyConstraintsForType(ref MixedRealityTransform transform, bool i
+
+        private void EnsureNewConstraintsInitialized()
+        {
+            var currentConstraints = host.GetComponents<TransformConstraint>();
```
Do you know what the calling mechanics are of EnsureNewConstraintsInitialized() - i.e. potentially on every single frame update?

Because if so, then the costs over time will probably not be that good - we'll end up doing component find/enumerate and then run a non-hashed array find on every single frame.

I think that we had a similar situation with some tap to place or solver stuff, where instead the constraints, when they were added to an object, would look for the SolverManager or something and be like "yo, I"m here now". You'd have to also make sure that the SolverManager/ConstraintManager would need to do a one-time enumeration when it started up, so that it could catch cases where individual constraints/solvers were added before the manager got added.

thalbern commented:
/azp run

thalbern commented:
changed the constraint manager to component and added register / unregister methods for runtime registering of components
also added hashset instead of the list that was there for keeping track of the components
integrated constraints now into inspector as well

thalbern commented:
/azp run

thalbern commented:
/azp run
